### PR TITLE
[MyMeta] Dashboard

### PR DESCRIPTION
**Pull request by vidvidvid**
## Overview

**What features/fixes does this PR include?**

MyMeta Dashboard.

## To-do

- [ ] Grid layout
  - [x] wrapper component
    - [x] header
      - [x] title
      - [ ] cog for configuration
    - [ ] configuration modals
    - [x] content
  - [ ] mobile responsive
  - [x] save it to localStorage
  - [ ] implement it on the backend

- [ ] Sections
  - [ ] Latest news
    - [ ] read
    - [ ] listen
    - [ ] watch
  - [ ] XP
    - [ ] total
    - [ ] ratio per month
    - [ ] xp
    - [ ] day/week/month/total
    - [ ] graph
  - [ ] Seeds
    - [ ] Market price
    - [ ] 24h trading volume
    - [ ] 24h low/high
  - [ ] Calendar
  - [ ] Leaderboard

## Possible improvements after implementing the mvp:

- [ ] Look into using `forwardRef` for making even more compact `GridItem`. [reference](https://github.com/MetaFam/TheGame/pull/885#pullrequestreview-793411859)

## Assets and Resources

[Notion](https://www.notion.so/meta-game/MyMeta-Dashboard-3c242de57e60488c952f424b5cdee931)
[Figma](https://www.figma.com/file/M3BpWO13WJVp7YivScL6K9/Dashboard?node-id=86%3A107)
[Original issue](https://github.com/MetaFam/TheGame/issues/337#event-5473742116)

Paired With: @luxumbra

### CODE REVIEW

**Comment by vidvidvid** [comment]
redeploy?

**Comment by vidvidvid** [comment]
Based on a discussion with @alalonde we decided to merge this pr into develop and incrementally continue merging features to develop instead of feature/dashboard branch, as we did with the past 3 prs.
This PR has 3 main features:
Feature/dashboard UI mvp (WIP) (#882) - set up the shell of the dashboard
Feature: Container Queries (#884) - implement responsiveness of dashboard grid boxes based on their size
Dashboard structure (#885) - refactor the structure of components
Also clicking the MetaGame icon in top left corner redirects the user to `/dashboard`, which shouldn't be a problem for now, since we will continually work on the dashboard and finish it relatively soon. If develop is going to be published to production, we can always change the route to `/`.

**Comment by alalonde** [comment]
> Also clicking the MetaGame icon in top left corner redirects the user to `/dashboard`, which shouldn't be a problem for now, since we will continually work on the dashboard and finish it relatively soon. If develop is going to be published to production, we can always change the route to `/`.

I'm always skeptical of time predictions, especially for large features like this... Let's just change it to `/` in the next PR

**Comment by vidvidvid** [comment]
> I'm always skeptical of time predictions, especially for large features like this... Let's just change it to `/` in the next PR

I can change it in this PR.

**Comment by vidvidvid** [comment]
@alalonde done - you can go ahead and merge it to develop whenever you find time 👍

File: packages/web/pages/dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+          <GridItem title="Leaderboard" sx={gridConfig.leaderboard}>
+            <Leaderboard />
+          </GridItem>
+        </Box>
+      </ResponsiveGridLayout>
+    </Box>
+  );
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```
Comment (alalonde) [inline packages/web/pages/dashboard.tsx:223]: Is this just for expediency until this is persisted to the player profile?

**Review by alalonde — APPROVED** [review]

**Pull request merged**